R/computeAnnualDegreeDays.R

```diff
@@ -1,3 +1,94 @@
+#' Initialize Degree Days Calculation
+#'
+#' Initiates the calculation of degree days for a single scenario, model, and time period.
+#' The calculation is split into yearly periods to improve computational stability.
+#'
+#' @param fileMapping A named list containing file paths and metadata. Expected elements include:
+#'   \describe{
+#'     \item{tas}{Path to temperature data file.}
+#'     \item{rsds}{Path to solar radiation data file (required if \code{bait} is TRUE).}
+#'     \item{sfcwind}{Path to surface wind data file (required if \code{bait} is TRUE).}
+#'     \item{huss}{Path to humidity data file (required if \code{bait} is TRUE).}
+#'     \item{start}{Start year of the time period.}
+#'     \item{end}{End year of the time period.}
+#'     \item{rcp}{RCP scenario identifier.}
+#'     \item{gcm}{GCM model identifier.}
+#'   }
+#' @param pop \code{SpatRaster} with annual population data.
+#' @param ssp \code{character} SSP scenario.
+#' @param bait \code{logical} indicating whether to use raw temperature or BAIT as ambient temperature.
+#' @param tLim \code{numeric} Temperature limits for degree day calculations.
+#' @param hddcddFactor \code{data.frame} containing pre-computed degree days.
+#' @param wBAIT \code{numeric} (Optional) Weights for BAIT adjustments. Default is \code{NULL}.
+#'
+#' @returns \code{data.frame} containing annual degree days.
+#'
+#' @author Hagen Tockhorn
+#'
+#' @importFrom dplyr mutate
+#' @importFrom magrittr %>%
+
+initCalculation <- function(fileMapping,
+                            pop,
+                            ssp,
+                            bait,
+                            tLim,
+                            hddcddFactor,
+                            wBAIT = NULL) {
+  # extract filenames
+  ftas  <- fileMapping[["tas"]]
+  frsds <- if (bait) fileMapping[["rsds"]]    else NULL
+  fsfc  <- if (bait) fileMapping[["sfcwind"]] else NULL
+  fhuss <- if (bait) fileMapping[["huss"]]    else NULL
```

Comment:
This looks as if it should be one object
```suggestion
fileNames <- c(ftas = fileMapping[["tas"]])
if (bait) {
  fileNames <- c(
    fileNames,
    frsds = fileMapping[["rsds"]]
    fsfc  = fileMapping[["sfcwind"]]
    fhuss = fileMapping[["huss"]]
  )
}
```